backends/arm/test/misc/test_debug_feats.py

```diff
@@ -110,18 +110,13 @@ def test_numerical_diff_prints(self):
             .partition()
             .to_executorch()
         )
-        if common.TOSA_REF_MODEL_INSTALLED:
-            # We expect an assertion error here. Any other issues will cause the
-            # test to fail. Likewise the test will fail if the assertion error is
-            # not present.
-            try:
-                # Tolerate 0 difference => we want to trigger a numerical diff
-                tester.run_method_and_compare_outputs(atol=0, rtol=0, qtol=0)
-            except AssertionError:
-                pass  # Implicit pass test
-            else:
-                self.fail()
+        # We expect an assertion error here. Any other issues will cause the
+        # test to fail. Likewise the test will fail if the assertion error is
+        # not present.
+        try:
+            # Tolerate 0 difference => we want to trigger a numerical diff
+            tester.run_method_and_compare_outputs(atol=0, rtol=0, qtol=0)
+        except AssertionError:
+            pass  # Implicit pass test
```

Comment:
Now when the TOSA tools are not installed these will just "pass" if someone tries to run these unit-tests - do you think it can be misleading compared to an explicit warning message before?

Reply:
We might need to check the logic here, good catch!

> This is because we won't check and skip tests for most pytest when TOSA tools are not installed? However, for the pull.yaml there shouldn't be any speed up for arm unit-tests, is that accurate?

Yes the idea is that we only run it by "hand" for now and if you have not done the proper setup to install TOSA/VELA/FVP  you will get an error, before we tried to just skip/ignore errors and 1. needed a lot of checks 2. could miss testing if stuff was not installed as it would seem fine :)

Or at least that is the idea, need to double check the negative case you found here. :)